tests/tests/htlc_tests.cpp

```diff
@@ -177,6 +177,448 @@ try {
+      trx.clear();
+   }
+   // Bob attempts to put a proposal on the blockchain of an HTLC with currency that cannot be transferred
+   // Does not fail before or after HF BSIP64
+   {
+      graphene::chain::htlc_create_operation create_operation;
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 3 coins to Bob");
+      create_operation.amount = graphene::chain::asset( 3, uia_id );
+      create_operation.to = joker_id;
+      create_operation.claim_period_seconds = 60;
+      create_operation.preimage_hash = hash_it<fc::sha256>( pre_image );
+      create_operation.preimage_size = preimage_size;
+      create_operation.from = bob_id;
+      create_operation.fee = db.get_global_properties().parameters.current_fees->calculate_fee(create_operation);
+      graphene::chain::proposal_create_operation prop_create;
+      prop_create.expiration_time = db.head_block_time() + 100;
+      prop_create.fee_paying_account = bob_id;
+      prop_create.review_period_seconds = 50;
+      prop_create.proposed_ops.emplace_back(create_operation);
+      trx.operations.push_back(prop_create);
+      sign(trx, alice_private_key);
+      PUSH_TX(db, trx, ~0);
+      trx.clear();
+   }
+
+   BOOST_TEST_MESSAGE("Fast Forwarding to HF BSIP64");
+   // fast forward to after the hardfork
+   generate_blocks( HARDFORK_CORE_BSIP64_TIME + 60 );
+   set_expiration( db, trx );
+
+   // Alice attempts to put a contract on the blockchain that is correct (and preimage size of 0)
+   {
+      graphene::chain::htlc_create_operation create_operation;
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 3 coins to Bob");
+      create_operation.amount = graphene::chain::asset( 3 * GRAPHENE_BLOCKCHAIN_PRECISION );
+      create_operation.to = bob_id;
+      create_operation.claim_period_seconds = 60;
+      create_operation.preimage_hash = hash_it<fc::hash160>( pre_image );
+      create_operation.preimage_size = 0;
+      create_operation.from = alice_id;
+      create_operation.extensions.value.memo = memo_data();
+      create_operation.extensions.value.memo->from = alice_public_key;
+      create_operation.extensions.value.memo->to = bob_public_key;
+      create_operation.extensions.value.memo->set_message(
+            alice_private_key, bob_public_key, "Dear Bob,\n\nMoney!\n\nLove, Alice");
+      create_operation.fee = db.get_global_properties().parameters.current_fees->calculate_fee(create_operation);
+      trx.operations.push_back(create_operation);
+      sign(trx, alice_private_key);
+      PUSH_TX(db, trx, ~0);
+      trx.clear();
+      graphene::chain::signed_block blk = generate_block();
+      processed_transaction alice_trx = blk.transactions[0];
+      alice_htlc_id = alice_trx.operation_results[0].get<object_id_type>();
+      generate_block();
+   }
+
+   // verify funds on hold... 100 - 3 = 97, minus the transaction fee = 90.57813
+   BOOST_CHECK_EQUAL( get_balance(alice_id, graphene::chain::asset_id_type()), 9057813 );
```

Comment:
Strictly speaking, from the available info in the test case, we don't know why the transaction fee is `97 - 90.57813`. Also there is no test case testing that changing kb fee of `transfer_operation` would impact fees for HTLC. We don't write test case for everything though.